### PR TITLE
[WIP] Training helpler

### PR DESCRIPTION
**Pull request by truong-my-mulodo**
## Task list

- [x] Constraints routes

- [x]  Add image in article

### CODE REVIEW

File: app/views/layouts/application.html.slim [hunk 1]
```diff
@@ -46,7 +46,7 @@ html
         li
           = link_to("Post", posts_path)
         li
-          = link_to("Articles", articles_path)
+          = link_to("Articles", user_articles_path(current_user))
```
Comment (nguyen-dat-mulodo) [inline app/views/layouts/application.html.slim]: current_user is nil, it cause error.
Please check it again.

File: config/routes.rb [hunk 2]
```diff
@@ -6,10 +6,14 @@
     get 'welcome/index'
     resources :articles
     root to: 'posts#index'
-    get 'user/articles/:user_id', to: 'articles#show_articles_by_user', as: 'show_articles_by_user'
+    # get 'user/articles/:user_id', to: 'articles#show_articles_by_user', as: 'show_articles_by_user'
```
Comment (nguyen-dat-mulodo) [inline config/routes.rb]: Please remove unused code.

File: config/routes.rb [hunk 3]
```diff
@@ -6,10 +6,14 @@
...
     devise_for :users
     resources :posts
     get '/welcome' => 'pages#welcome'
+    resources :users do
+      resources :articles, constraints: { id: /[0-9]+/ }
+
+    end
     # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
     # get "users/sign_out",to: "sessions#destroy"
```
Comment (nguyen-dat-mulodo) [inline config/routes.rb]: Please remove unused code.

Comment (truong-my-mulodo) [reply config/routes.rb]: I cant remove it, that is code's someone. Please check it, Thanks

**Comment by nguyen-dat-mulodo** [comment]
I reviewed it. Thanks
@tran-hai-mulodo . Please check it again and merge it.

**Comment by tran-hai-mulodo** [comment]
@truong-my-mulodo
cc: @nguyen-dat-mulodo 

I reviewed. It's OK now. Thanks for your hard work.

**Pull request merged**